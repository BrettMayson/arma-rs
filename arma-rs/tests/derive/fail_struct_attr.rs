use arma_rs_proc::{FromArma, IntoArma};

#[derive(IntoArma, FromArma)]
#[arma]
struct NoList;

#[derive(IntoArma, FromArma)]
#[arma("literal")]
struct Literal;

#[derive(IntoArma, FromArma)]
#[arma(unknown)]
struct Unknown;

#[derive(IntoArma, FromArma)]
#[arma(unknown::path)]
struct UnknownPath;

#[derive(IntoArma, FromArma)]
#[arma(duplicate, duplicate)]
struct Duplicate;

#[derive(IntoArma, FromArma)]
struct FieldUnknown {
    #[arma(unknown)]
    test: u32,
}

#[derive(IntoArma, FromArma)]
struct FieldDuplicate {
    #[arma(duplicate, duplicate)]
    test: u32,
}

fn main() {}
