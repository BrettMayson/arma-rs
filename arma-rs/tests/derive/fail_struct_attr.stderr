error: expected attribute arguments in parentheses: #[arma(...)]
 --> tests/derive/fail_struct_attr.rs:4:3
  |
4 | #[arma]
  |   ^^^^

error: unexpected literal in nested attribute, expected ident
 --> tests/derive/fail_struct_attr.rs:8:8
  |
8 | #[arma("literal")]
  |        ^^^^^^^^^

error: unknown arma-rs container attribute `unknown`
  --> tests/derive/fail_struct_attr.rs:12:8
   |
12 | #[arma(unknown)]
   |        ^^^^^^^

error: unknown arma-rs container attribute `unknown::path`
  --> tests/derive/fail_struct_attr.rs:16:8
   |
16 | #[arma(unknown::path)]
   |        ^^^^^^^^^^^^^

error: duplicate arma-rs attribute `default`
  --> tests/derive/fail_struct_attr.rs:20:17
   |
20 | #[arma(default, default)]
   |                 ^^^^^^^

error: unknown arma-rs field attribute `unknown`
  --> tests/derive/fail_struct_attr.rs:25:12
   |
25 |     #[arma(unknown)]
   |            ^^^^^^^

error: duplicate arma-rs attribute `default`
  --> tests/derive/fail_struct_attr.rs:31:21
   |
31 |     #[arma(default, default)]
   |                     ^^^^^^^
